datafusion/physical-expr/src/aggregate/count_distinct.rs

```diff
@@ -41,7 +41,7 @@ pub struct DistinctCount {
     /// The DataType for the final count
     data_type: DataType,
     /// The DataType used to hold the state for each input
-    state_data_types: Vec<DataType>,
+    state_data_types: DataType,
     /// The input arguments
     exprs: Vec<Arc<dyn PhysicalExpr>>,
```

Comment:
I think we should also update `exprs: Vec<Arc<dyn PhysicalExpr>>` to `exprs:  Arc<dyn PhysicalExpr>` and `DistinctCount::new` to communicate that multiple columns are no longer supported.

Reply:
Maybe `input_data_types: Vec<DataType>` also need to be changed to `DataType`?

Reply:
Yeah, as well.
If you like to do some more refactoring, you could remove `data_type: DataType` from the `DistinctCount` struct as well and just keep a `int64` in the `DistinctCountAccumulator` instead of using a scalar value.